packages/core/src/abap/5_syntax/basic_types.ts

```diff
@@ -125,6 +125,10 @@ export class BasicTypes {
       return new Types.NumericGenericType();
     } else if (chainText === "UTCLONG") { // todo, take version into account
       return new Types.UTCLongType();
+    } else if (chainText === "DECFLOAT16") {
+      return new Types.DecFloat16Type();
+    } else if (chainText === "DECFLOAT32") {
```

Comment:
Should be `DECFLOAT34`, no?

Reply:
yea, thanks

Reply:
fixed in https://github.com/abaplint/abaplint/pull/1433/commits/a2b83fb047ffd41d84ca021bd81200e07b38b796